agent/structs/config_entry_gateways.go

```diff
@@ -99,14 +100,29 @@ type GatewayTLSConfig struct {
 
 	// SDS allows configuring TLS certificate from an SDS service.
 	SDS *GatewayTLSSDSConfig `json:",omitempty"`
+
+	TLSMinVersion types.TLSVersion `json:",omitempty"`
+	TLSMaxVersion types.TLSVersion `json:",omitempty"`
```

Comment:
Given that we have a zero value that represents a meaningful state (`TLSVersionInvalid`), do we want to omit it from JSON if we ever marshal it?

I think it would be important for the user to explicitly see the field having a zero value and correct it.

Reply:
Seems strange to me that the default zero-value would be `TLSVersionInvalid` -- which arguably should mean "you set an invalid value for the TLS version" rather than "you didn't set a TLS version" (which I would normally think would mean, "default to our previous auto-selection behavior") -- if users need to "correct" a default state (since "correct" implies an error) tells me that either:

1. it shouldn't be the default state
2. if it is adopted as the default state it's essentially a breaking change

Unless I'm missing some context due to my lack of familiarity with the core codebase?

Reply:
Added `TLSVersionUnspecified` as a distinct valid zero-value over in https://github.com/hashicorp/consul/pull/11645 after chatting with @andrewstucki